ffmpeg_tools/formats.py

```diff
@@ -60,17 +81,90 @@ def is_supported(vformat):
     def list_supported_formats(vformat):
         return list_supported_formats()
 
+    def get_demuxer(self):
+        return _DEMUXER_MAP[self].value
+
+    def is_exclusive_demuxer(self):
+        return self in _EXCLUSIVE_DEMUXERS
+
+    def get_matching_muxers(self):
+        return {
+            muxer
+            for muxer, demuxer in _DEMUXER_MAP.items()
+            if demuxer == self
+        }
+
+
+# This set containe demuxers that cannot be used as muxers in ffmpeg.
+_EXCLUSIVE_DEMUXERS = {
+    Container.c_QUICK_TIME_DEMUXER,
+    Container.c_MATROSKA_WEBM_DEMUXER,
+}
 
 
-_QUICKTIME_CODECS = {
-    "videocodecs": ["h264", "h265", "HEVC", "mpeg1video", "mpeg2video"],
-    "audiocodecs": ["mp3", "aac"]
+# This dictionary defines which ffmpeg demuxer handles files encoded by which
+# muxer.
+_DEMUXER_MAP = {
+    Container.c_3G2: Container.c_QUICK_TIME_DEMUXER,
+    Container.c_3GP: Container.c_QUICK_TIME_DEMUXER,
+    Container.c_AVI: Container.c_AVI,
```

Comment:
If muxer and demuxer is the same, we shouldn't add it to this map. It will grow fast than. Better approach would be to store only exceptions here and assume rest muxers and demuxers are the same

Reply:
Done.

I had to extend the function that gives you a list of muxers handled by a demuxer because it now requires additional logic:
- It has to know for example that even if we have a `c_SVCD: c_MPEG` pair defined it still has to add a `c_MPEG: c_MPEG` pair because `mpeg` demuxer handles two muxers: itself and `svcd`.
- It has to know that `c_QUICK_TIME_DEMUXER` is not a muxer so there should be no pair for it.

It was a bit more straightforward when defined manually but this works too thanks to the `_EXCLUSIVE_DEMUXERS` set.

Also, by not having these pairs explicitly we lose a tiny bit of flexibility. For example it's now not possible to have the `svcd`/`mpeg` pair without `mpeg`/`mpeg` pair. But it seems like an unlikely case so we probably don't need this flexibility anyway.